Documentation/BasicOperators.md

```diff
@@ -46,41 +46,394 @@ code-style will be used.
+    })
+```
+
+Note, that nothing will be printed until `producer` is started (possibly somewhere else).
 
 ## Operator composition
 
-### Lifting
 ### Pipe
 
+The `|>` operator can be used to apply a signal operator to a signal. Multiple operators can be chained after each other using the `|>` operator
+
+```Swift
+intSignal
+    |> filter { num in num % 2 == 0 }
+    |> map(toString)
+    |> observe(next: { string in println(string) })
+```
+
+### Lifting
+
+Signal operators can be _lifted_ to operate upon `SignalProducer`s instead using the `lift` operator.
+In other words, this will create a new `SignalProducer` which will apply the given signal operator to _every_ signal created from the incoming `SignalProducer`s
+just if the operator had been applied to each signal yielded from `start()`.
+
+The `|>` operator implicitly lifts signal operators, when used with `SignalProducer`s.
 
 ## Transforming signals
 
+These operators transform a signal into a new signal.
+
 ### Mapping
+
+The `map` operator is used to transform the values in a signal, creating a new signal with the results.
+
+```Swift
+let (signal, sink) = Signal<String, NoError>.pipe()
+signal
+    |> map { string in string.uppercaseString }
+    |> observe(next: { println($0) })
+
+sendNext(sink, "a")     // Prints A
+sendNext(sink, "b")     // Prints B
+sendNext(sink, "c")     // Prints C
+```
+
+
 ### Filtering
+
+The `filter` operator is used to include only values in a signal that satisfy a predicate
+
+```Swift
+let (signal, sink) = Signal<Int, NoError>.pipe()
+signal
+    |> filter { number in number % 2 == 0 }
+    |> observe(next: { println($0) })
+
+sendNext(sink, 1)     // Not printed
+sendNext(sink, 2)     // Prints 2
+sendNext(sink, 3)     // Not printed
+sendNext(sink, 4)     // prints 4
+```
+
 ### Reducing
 
+The `reduce` operator is used to aggregate a signals value into a single combine value. Note, that the final value is only sent after the source signal completes.
+
+```Swift
+let (signal, sink) = Signal<Int, NoError>.pipe()
+
+signal
+    |> reduce(1) { $0 * $1 }
+    |> observe(next: { println($0) })
+
+sendNext(sink, 1)     // nothing printed
+sendNext(sink, 2)     // nothing printed
+sendNext(sink, 3)     // nothing printed
+sendCompleted(sink)   // prints 6
+```
+
 
 ## Combining signals
 
+These operators combine multiple signals into a single new signal.
+
 ### Combining latest values
+
+The `combineLatest` function combines the latest values of two (or more) signals. The resulting signal will only send a the first value after both inputs have sent at least one value each. After that, each value on either of the inputs will cause a new value on the output.
+
+```Swift
+let (numbersSignal, numbersSink) = Signal<Int, NoError>.pipe()
+let (lettersSignal, lettersSink) = Signal<String, NoError>.pipe()
+
+combineLatest(numbersSignal, lettersSignal)
+    |> observe(next: { println($0) },
+          completed: { println("Completed") })
```

Comment:
can you single line this - `|> observe(next: println, completed: { println("Completed") })`